checkbox-support/checkbox_support/parsers/v4l2_compliance.py

```diff
@@ -0,0 +1,207 @@
+    """
+    assert line.startswith(
+        "test"
+    ), "This line doesn't describe a test output. Line is {}".format(line)
+    test_name = line.split("test ", maxsplit=1)[1].split(": ", maxsplit=1)[0]
+    return test_name, test_name.startswith("VIDIOC")
+
+
+def parse_v4l2_compliance(
+    device: T.Union[int, str, None] = None,
+) -> T.Tuple[Summary, Details]:
+    """Parses the output of v4l2-compliance
+
+    :param device: which device to test, defaults to "/dev/video0",
+    it can also be an integer. See v4l2-compliance -h
+    :type device: T.Union[int, str], optional
+    :return: 2 dictionaries (summary, details).
+    NOTE: summary comes from directly parsing the numbers in the last line of
+    v4l2-compliance and it does **NOT** match the array sizes in Details
+    since we map the test names to actual ioctls.
+
+    :rtype: T.Tuple[Summary, Details]
+    """
+
+    out = sp.run(
+        ["v4l2-compliance", *(["-d", str(device)] if device else [])],
+        universal_newlines=True,
+        stdout=sp.PIPE,
+        stderr=sp.PIPE,
+    )  # can't really depend on the return code here
+    # since any failure => return code 1
+
+    error_prefixes = ("Failed to open", "Cannot open device")
+    if any(out.stderr.startswith(prefix) for prefix in error_prefixes):
+        # can't open the device
+        raise FileNotFoundError(out.stderr)
+
+    lines = [line.strip() for line in out.stdout.splitlines() if line.strip()]
+
+    pattern = (
+        r"Total(?: for (.*))?: (.*), "
+        r"Succeeded: (.*), Failed: (.*), Warnings: (.*)"
+    )
+    match_output = re.match(pattern, lines[-1])
+
+    assert match_output is not None, (
+        "There's no summary line in v4l2-compliance's output. "
+        "Output might be corrupted. Last line is: \n {}".format(lines[-1])
+    )
+
+    summary = {
+        "device_name": match_output.group(1),  # could be None, but won't panic
+        "total": int(match_output.group(2)),
+        "succeeded": int(match_output.group(3)),
+        "failed": int(match_output.group(4)),
+        "warnings": int(match_output.group(5)),
+    }
+
+    details = {
+        "succeeded": [],
+        "failed": [],
+        "not_supported": [],
+    }  # type: dict[str, list[str]]
+
+    for line in lines:
+        if summary["device_name"] is None and line.startswith(
+            "Compliance test for"
+        ):
+            # try to see if there's a device name at the top
+            # if not, report None
+            name_line_pattern = r"Compliance test for (.*)"
+            name_match = re.match(name_line_pattern, line)
+            if name_match is not None:
+                clean_name = (
+                    name_match.group(1)
+                    .replace(":", "")
+                    .replace("(not using libv4l2)", "")
+                    .strip()
+                )
+                if clean_name != "":
+                    summary["device_name"] = clean_name
+            continue
```

Comment:
The regex is not needed here, match will always match because the line startswith the match and `.*` always matches, regardless, we can do something way simpler here:
```suggestion
        if not summary["device_name"] and line.startswith(
            "Compliance test for"
        ):
            # try to see if there's a device name at the top
            # if not, report None
            clean_name = (
                line
                .replace("Compliance test for ", "")
                .replace(":", "")
                .replace("(not using libv4l2)", "")
                .strip()
            )
            if clean_name:
                summary["device_name"] = clean_name
            continue
```